src/traces/pie/plot.js

```diff
@@ -454,6 +501,72 @@ function transformOutsideText(textBB, pt) {
     };
 }
 
+function positionTitleInside(titleBB, cd0) {
+    var textDiameter = Math.sqrt(titleBB.width * titleBB.width + titleBB.height * titleBB.height);
+    return {
+        x: cd0.cx,
+        y: cd0.cy,
+        scale: cd0.trace.hole * cd0.r * 2 / textDiameter
+    };
+}
+
+function positionTitleOutside(titleBB, cd0, plotSize) {
+    var scaleX, scaleY, chartWidth, titleSpace, titleShift, maxPull;
+    var trace = cd0.trace;
+
+    maxPull = getMaxPull(trace);
+    chartWidth = plotSize.w * (trace.domain.x[1] - trace.domain.x[0]);
+    scaleX = chartWidth / titleBB.width;
+    if(isSinglePie(trace)) {
+        titleShift = trace.titlefont.size / 2;
+        // we need to leave enough free space for an outside label
+        if(trace.outsidetextfont) titleShift += 3 * trace.outsidetextfont.size / 2;
+        else titleShift += trace.titlefont.size / 4;
+        return {
+            x: cd0.cx,
+            y: cd0.cy - (1 + maxPull) * cd0.r - titleShift,
+            scale: scaleX
+        };
+    }
+    titleSpace = getTitleSpace(trace, plotSize);
+    // we previously left a free space of height titleSpace.
+    // The text must fit in this space.
+    scaleY = titleSpace / titleBB.height;
+    return {
+        x: cd0.cx,
+        y: cd0.cy - (1 + maxPull) * cd0.r - (titleSpace / 2),
+        scale: Math.min(scaleX, scaleY)
+    };
+}
+
+function isSinglePie(trace) {
+    // check if there is a single pie per y-column
+    if(trace.domain.y[0] === 0 && trace.domain.y[1] === 1) return true;
```

Comment:
I don't think we want to treat single-pie-per-column differently from multiple, but it's not clear to me offhand whether to shrink all or none. The argument for shrinking all of them is that there's also `layout.title` at the top, and we don't want pie titles to overlap with that. The argument for shrinking none is that we already put other labels in the margins - cartesian axis  titles, ticks, & tick labels. People making multi-subplot plots are used to leaving space for this between their subplots, and in fact if you make a `layout.grid` (see https://codepen.io/alexcjohnson/pen/NBZYpm/) there's a gap by default between neighboring subplots, which is easy to increase (`layout.grid.ygap`).

I guess *most* of our subplots - whether single-trace like pie or multi-trace like cartesian - stay within their domains, which would argue for shrinking all. The only exceptions I think are cartesian and ternary subplots.

Reply:
> The only exceptions I think are cartesian and ternary subplots.

Polar axes are also drawn out-of-their domains (for most cases that only implies angularaxis tick and tick labels). In other words only the _plotting_ area is constraint by the domain.

If I understand correctly, long outside labels (e.g. https://rreusser.github.io/plotly-mock-viewer/#pie_style_arrays) do not have an effect on the margins, but arguably _could_ (just like cartesian axis tick labels now have an `automargin:true` option). 

All in all, I'm leaning on the side on consistency with outside labels, that is don't touch the margins but shrink the pies.



> I don't think we want to treat single-pie-per-column differently from multiple

I agreed :100: % here. So yeah, let's shrink them all!

Reply:
Done. I always shrink the pies to gain space for the title.